op-service/log/cli.go

```diff
@@ -46,28 +63,14 @@ type CLIConfig struct {
 	Format string // Format the log output. Supported formats: 'text', 'terminal', 'logfmt', 'json', 'json-pretty'
 }
 
-func (cfg CLIConfig) Check() error {
-	switch cfg.Format {
-	case "json", "json-pretty", "terminal", "text", "logfmt":
-	default:
-		return fmt.Errorf("unrecognized log format: %s", cfg.Format)
-	}
-
-	level := strings.ToLower(cfg.Level)
-	_, err := log.LvlFromString(level)
-	if err != nil {
-		return fmt.Errorf("unrecognized log level: %w", err)
+func NewLogger(ctx *cli.Context, cfg CLIConfig) log.Logger {
+	var wr io.Writer = os.Stdout
+	if ctx != nil && ctx.App != nil {
+		wr = ctx.App.Writer
 	}
-	return nil
-}
-
-func NewLogger(cfg CLIConfig) log.Logger {
-	handler := log.StreamHandler(os.Stdout, Format(cfg.Format, cfg.Color))
+	handler := log.StreamHandler(wr, Format(cfg.Format, cfg.Color))
 	handler = log.SyncHandler(handler)
 	handler = log.LvlFilterHandler(Level(cfg.Level), handler)
-	// Set the root handle to what we have configured. Some components like go-ethereum's RPC
-	// server use log.Root() instead of being able to pass in a log.
-	log.Root().SetHandler(handler)
```

Comment:
This was pretty explicitly set. Because we use geth as a fully fledged library here, we need to adapt to their logging patterns. I'm not aware of a way to provide  a logger to the RPC component. See https://github.com/ethereum-optimism/optimism/pull/5086

Setting the root handler is also beneficial in the case where we accidentally call `log.Info` instead of `logger.Info`. I do acknowledge that it has a tradeoff when running multiple in memory instances and is not nice to have a global logger.

Reply:
I changed it so that the global-log-handler is explicitly set to a copy of the created log handler during the CLI setup phase, this way the logging setup is otherwise side-effect free, while we still preserve the global log output in regular program execution. And then during op-e2e testing we can set the global log handler to the testlog handler, without it being overwritten, cleaning up the test logs nicely.